lighthouse-core/gather/gather-runner.js

```diff
@@ -96,7 +96,8 @@ class GatherRunner {
       .then(_ => driver.enableRuntimeEvents())
       .then(_ => driver.evaluateScriptOnLoad('window.__nativePromise = Promise;'))
       .then(_ => driver.cleanAndDisableBrowserCaches())
-      .then(_ => driver.clearDataForOrigin(options.url));
+      .then(_ => driver.clearDataForOrigin(options.url))
+      .then(_ => driver.blockUrlPatterns(options.flags.blockedUrlPatterns || []));
```

Comment:
I was mulling over if `blockedUrlPatterns` should live right on `options` rather than `options.flags`, but it'd probably require a line of plumbing in `index.js` that I don't love.

So yeah, I'm also fine with it hanging out on `options.flags` even if its not user-exposed.